### PR TITLE
remove items from sequence

### PR DESCRIPTION
**Pull request by StarJiao**
based on pull request 471 [pullrequest 471](https://github.com/jbeder/yaml-cpp/pull/471)
allow users remove items from sequence and keep the type

### CODE REVIEW

File: util/demo.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#include "yaml-cpp/yaml.h"
```
Comment (jbeder) [inline util/demo.cpp]: Instead of a new demo executable, unit tests are sufficient.

File: test/node/node_test.cpp [hunk 2]
```diff
@@ -106,6 +130,14 @@ TEST(NodeTest, RemoveUnassignedNode) {
   EXPECT_EQ(0, node.size());
 }
 
+TEST(NodeTest, RemoveUnassignedNodeFromMap) {
```
Comment (jbeder) [inline test/node/node_test.cpp]: Similarly, please add a tests that verifies that if you have a map, where one of the keys is an integer, you can remove it and it's still a map.

File: include/yaml-cpp/node/detail/impl.h [hunk 3]
```diff
@@ -56,6 +56,39 @@ struct get_idx<Key, typename std::enable_if<std::is_signed<Key>::value>::type> {
   }
 };
 
+template <typename Key, typename Enable = void>
+struct remove_idx {
+  static char remove(std::vector<node*>& /* sequence */, const Key& /* key */) {
```
Comment (jbeder) [inline include/yaml-cpp/node/detail/impl.h]: No need to comment out the variables.

File: include/yaml-cpp/node/detail/impl.h [hunk 4]
```diff
@@ -129,23 +162,29 @@ inline node& node_data::get(const Key& key, shared_memory_holder pMemory) {
...
-
-  for (node_map::iterator it = m_map.begin(); it != m_map.end(); ++it) {
-    if (it->first->equals(key, pMemory)) {
-      m_map.erase(it);
-      return true;
-    }
-  }
+	if (m_type == NodeType::Sequence) {
+		char result = remove_idx<Key>::remove(m_sequence, key);
+		if (result == 0) return false;
```
Comment (jbeder) [inline include/yaml-cpp/node/detail/impl.h]: Add curly braces around all "if" statements, even if they're one line.

(The codebase isn't quite consistent there, but I'm trying to enforce it for new code.)

File: include/yaml-cpp/node/detail/impl.h (hunk 3 above)
Comment (jbeder) [inline include/yaml-cpp/node/detail/impl.h]: Why does this return a 'char'? It looks like in all implementations, it either returns 0 or 1.

File: include/yaml-cpp/node/detail/impl.h [hunk 5]
```diff
@@ -129,23 +162,29 @@ inline node& node_data::get(const Key& key, shared_memory_holder pMemory) {
...
+		if (result == 1) return true;
+		convert_to_map(pMemory);
```
Comment (jbeder) [inline include/yaml-cpp/node/detail/impl.h]: Can this ever happen? (See above comment.)

File: include/yaml-cpp/node/detail/impl.h [hunk 6]
```diff
@@ -129,23 +162,29 @@ inline node& node_data::get(const Key& key, shared_memory_holder pMemory) {
...
+	}
+	if (m_type == NodeType::Map) {
+
```
Comment (jbeder) [inline include/yaml-cpp/node/detail/impl.h]: Extra newline.

**Comment by StarJiao** [comment]
@jbeder Thank you so much and I'm honoured by your reviewing .To be honest, this is the first time to create a pull request and there is many rules i need learn.  I am not familiar with C++ . Haha. Thanks again and I will fix all these mistake by your review.

**Comment by jbeder** [comment]
No problem! Thanks for contributing!

**Comment by StarJiao** [comment]
@jbeder Done

**Pull request merged**